classes/pdf/PDF.php

```diff
@@ -192,4 +187,40 @@ public function getTemplateObject($object)
 
         return $class;
     }
+
+    /**
+     * Get the PDF filename based on the objects.
+     *
+     * @return string
+     */
+    public function getFilename(): string
+    {
+        if (empty($this->filename)) {
+            $this->setFilename();
+        }
+
+        return $this->filename;
+    }
+
+    /**
+     * Set the PDF filename based on the objects.
+     *
+     * @return bool
+     */
+    public function setFilename(): bool
+    {
+        foreach ($this->objects as $object) {
```

Comment:
Maybe I dont understand this method purpose but, everytime an object has template, it overrides the filename.

However, You can put the count outside of the foreach and do
$this->filename = ($count>1) ? $template->getBulkFilename() : $template->getFilename();
or sthg like that

Finally I don't see why you return a bolean for this method. The return value is used nowhere

Reply:
> Maybe I dont understand this method purpose but, everytime an object has template, it overrides the filename.

This only happens when no filename has been set yet.

> Finally I don't see why you return a bolean for this method. The return value is used nowhere

I added this since I saw it done in other setter methods as well, no other reason.

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/26587/commits/a6b421b9630f79353d29d67ae6d7ef96eb5a268b and https://github.com/PrestaShop/PrestaShop/pull/26587/commits/3a2aed27ff9a9c5906b84271a6a1c8152e1ce57b